pkg/network/common/egressip.go

```diff
@@ -541,13 +734,17 @@ func (eit *EgressIPTracker) findEgressIPAllocation(eip *egressIPInfo, allocation
 	bestNode := ""
 	otherNodes := false
 
-	for _, node := range eit.nodes {
-		if node.offline {
-			continue
-		}
-		if eit.nodeHasEgressIPForNamespace(node, eip, allocation) {
-			continue
-		}
+	// Given a capacity constraint allocation problem: we need to assign IPs to
+	// `bestNode` with the highest availability (capacity - current assignment).
+	// This is needed as to avoid sub-optimal allocations, meaning: avoiding
+	// assigning IPs to nodes with low availability first only to realize later
+	// that the remaining IPs can't be allocated to the nodes with higher
+	// availability because they violate other constraints, such as: already
+	// having another IP in that namespace allocated to it. Hence we need to
+	// sort nodes in increasing order of availability as to make sure the
+	// `bestNode` gets chosen as the node with the highest availability.
```

Comment:
The full constraint is that `bestNode` is the node with the highest availability, out of the set of nodes that are tied for having the fewest current egress IPs, out of the set of nodes that don't currently have an assignment for this namespace.
(Right?)

Reply:
Yes, that is correct. I updated the doc bloc with your phrasing.